test/functional/Java17andUp/src_170/org/openj9/test/jep389/upcall/UpcallMHWithMixedSigStruTests.java

```diff
@@ -0,0 +1,934 @@
+		VarHandle elemHandle2 = structLayout.varHandle(double.class, PathElement.groupElement("elem2"));
+
+		MethodType mt = MethodType.methodType(double.class, double.class, MemorySegment.class, MemoryAddress.class);
+		FunctionDescriptor fd = FunctionDescriptor.of(C_DOUBLE, C_DOUBLE, structLayout, C_POINTER);
+		Addressable functionSymbol = nativeLibLookup.lookup("addDoubleAndLongDoubleFromStructByUpcallMH").get();
+		MethodHandle mh = clinker.downcallHandle(functionSymbol, mt, fd);
+
+		try (ResourceScope scope = ResourceScope.newConfinedScope()) {
+			MemoryAddress upcallFuncAddr = clinker.upcallStub(UpcallMethodHandles.MH_addDoubleAndLongDoubleFromStruct,
+					FunctionDescriptor.of(C_DOUBLE, C_DOUBLE, structLayout), scope);
+			SegmentAllocator allocator = SegmentAllocator.ofScope(scope);
+			MemorySegment structSegmt = allocator.allocate(structLayout);
+			elemHandle1.set(structSegmt, 22222222222222L);
+			elemHandle2.set(structSegmt, 33333.444D);
+
+			double result = (double)mh.invokeExact(55555.111D, structSegmt, upcallFuncAddr);
+			Assert.assertEquals(result, 22222222311110.555D, 0.001D);
+		}
+	}
+
+	@Test
+	public void test_addFloatAndInt3FloatsFromStructByUpcallMH() throws Throwable {
+		GroupLayout structLayout = MemoryLayout.structLayout(C_INT.withName("elem1"),
+				C_FLOAT.withName("elem2"), C_FLOAT.withName("elem3"), C_FLOAT.withName("elem4"));
+		VarHandle elemHandle1 = structLayout.varHandle(int.class, PathElement.groupElement("elem1"));
+		VarHandle elemHandle2 = structLayout.varHandle(float.class, PathElement.groupElement("elem2"));
+		VarHandle elemHandle3 = structLayout.varHandle(float.class, PathElement.groupElement("elem3"));
+		VarHandle elemHandle4 = structLayout.varHandle(float.class, PathElement.groupElement("elem4"));
+
+		MethodType mt = MethodType.methodType(float.class, float.class, MemorySegment.class, MemoryAddress.class);
+		FunctionDescriptor fd = FunctionDescriptor.of(C_FLOAT, C_FLOAT, structLayout, C_POINTER);
+		Addressable functionSymbol = nativeLibLookup.lookup("addFloatAndInt3FloatsFromStructByUpcallMH").get();
+		MethodHandle mh = clinker.downcallHandle(functionSymbol, mt, fd);
+
+		try (ResourceScope scope = ResourceScope.newConfinedScope()) {
+			MemoryAddress upcallFuncAddr = clinker.upcallStub(UpcallMethodHandles.MH_addFloatAndInt3FloatsFromStruct,
+					FunctionDescriptor.of(C_FLOAT, C_FLOAT, structLayout), scope);
+			SegmentAllocator allocator = SegmentAllocator.ofScope(scope);
+			MemorySegment structSegmt = allocator.allocate(structLayout);
+			elemHandle1.set(structSegmt, 77777777);
+			elemHandle2.set(structSegmt, 11.22F);
+			elemHandle3.set(structSegmt, 22.33F);
+			elemHandle4.set(structSegmt, 44.55F);
+
+			float result = (float)mh.invokeExact(66.678F, structSegmt, upcallFuncAddr);
+			Assert.assertEquals(result, 77777921.778F, 0.001F);
+		}
+	}
+
+	@Test
+	public void test_addLongAndLong2FloatsFromStructByUpcallMH() throws Throwable {
```

Comment:
This test fails on Windows for both HotSpot and OpenJ9.  However, the problem seems to be on the "downcall" part of the test as the elements of the struct don't even reach the C function intact (i.e., before we can even do an upcall).

For HotSpot, printing the args reaching the C function (`addLongAndLong2FloatsFromStructByUpcallMH`) before the upcall:
```
IN C: addLongAndLong2FloatsFromStruct : arg1=555555555555, arg2.elem1=388697201, arg2.elem2=0.000000, arg2.elem3=11.250000
```
For OpenJ9, printing the args reaching the C function before the upcall:
```
IN C: addLongAndLong2FloatsFromStruct : arg1=555555555555, arg2.elem1=388697201, arg2.elem2=0.000000, arg2.elem3=11.250000
```

Reply:
@0xdaryl, could you help double-check `values[1]` or `ffiArgs[1]` for the struct the code of `inlInternalDowncallHandlerInvokeNative` in downcall at https://github.com/eclipse-openj9/openj9/blob/b9b733900ef25ba25435e8dd557a10f4e39a4fd6/runtime/vm/BytecodeInterpreter.hpp#L5103.  I'd like to know whether or not the passed-in value is correct given the test passes on other platforms.

If the passed-in struct is correct but it gets corrupted in the C native functions, then problem should originate from `libffi` on Windows in which case we can disable this test for the moment and will need to create an issue at https://github.com/libffi/libffi/issues or https://sourceware.org/mailman/listinfo/libffi-discuss to request them to fix it up.

Reply:
I suspect the problem was caused by the wrong element type defined for `elem1` in `stru_Long_2_Floats` at `downcall.h`
```
typedef struct stru_Long_2_Floats {
	long elem1;  <---------
	float elem2;
	float elem3;
} stru_Long_2_Floats;
```
in which `long` is 4 bytes on Windows, which should be replaced with `LONG` (int64_t). I will update this to verify again on Windows to see how it goes.

Reply:
I just verified with the updated definition of `stru_Long_2_Floats` in `/runtime/tests/clinkerffi/downcall.h`
```
typedef struct stru_Long_2_Floats {
	LONG elem1;  <-------
	float elem2;
	float elem3;
} stru_Long_2_Floats;
```
and the test `test_addLongAndLong2FloatsFromStructByUpcallMH` passed on both OpenJ9 and Hotspot:
```
$ java -version
openjdk version "17.0.5-internal" 2022-10-18
OpenJDK Runtime Environment (build 17.0.5-internal+0-adhoc.Administrator.openj9-openjdk-jdk17)
Eclipse OpenJ9 VM (build jep389_419_ffi_upcall_v4_struct_v2_outofline_mh_v9_fin_v8_5-c3c03e474,
 JRE 17 Windows 10 amd64-64-Bit Compressed References 20220920_000000 (JIT enabled, AOT enabled
)
OpenJ9   - c3c03e474
OMR      - b1be89cf5
JCL      - e988b7def79 based on jdk-17.0.5+5)

$ java -version
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-86)
OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)
...
PASSED: test_addLongAndLong2FloatsFromStructByUpcallMH
```

Reply:
@0xdaryl,  please double-check the test on your side (note that the test library `clinkerffitests.dll` needs to be regenerated with the latest `downcall.h` in this PR)